app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabase.kt

```diff
@@ -1082,10 +902,10 @@ class FirebaseDatabase : Database() {
         val future = CompletableFuture<Unit>()
 
         // the timestamp of the message is used as a key
-        val data = listOf<Pair<String, Any?>>(
-            "${FirebaseKeys.CHAT_MESSAGE_CONTENT_TEXT}" to (firstMessage.content as MessageContent.Text).text,
-            "${FirebaseKeys.CHAT_MESSAGE_SENDER}" to firstMessage.senderId,
-        ).filter { it.second != null }.associate { entry -> entry }
+        val data = mapOf(
+            FirebaseKeys.CHAT_MESSAGE_CONTENT_TEXT to (firstMessage.content as MessageContent.Text?)?.text,
+            FirebaseKeys.CHAT_MESSAGE_SENDER to firstMessage.senderId,
```

Comment:
The face can't be null (I don't think that the two ?)? are necessary)

Reply:
What do you mean by face? 

If you talk about `(firstMessage.content as MessageContent.Text?)?.text` then we can imagine the function receiving another content type.